packages/admin-panel/src/VizBuilderApp/vizBuilderConfigStore.js

```diff
@@ -10,22 +10,27 @@ import React, { useReducer, createContext, useContext, useCallback } from 'react
  * possible in the component tree.
  */
 const initialConfigState = {
-  name: null,
-  summary: null,
   project: null,
   location: null,
-  permissionGroup: null,
-  data: {
-    dataElements: [],
-    dataGroups: [],
-    aggregations: [],
-    transform: [],
+  testData: null,
+  visualisation: {
+    id: null,
+    name: null,
+    code: null,
+    permissionGroup: null,
+    data: {
+      dataElements: [],
+      dataGroups: [],
+      aggregations: [],
+      transform: [],
+    },
+    presentation: {},
```

Comment:
Change the store's structure a bit to allow storing some extra info and avoid affecting the visualisation object